Refactor function names of `rustc_ast_lowering`

reddevilmidzy opened the pull request:
close: rust-lang/rust#150062 

I'll rebase after https://github.com/rust-lang/rust/pull/149114 is merged.

Would it be better to change `lower_const_path_to_const_arg` → `lower_const_path_to_const_arg_and_intern`,
`lower_array_length_to_const_arg` → `lower_array_length_to_const_arg_and_intern`?


r? @lcnr

lcnr reviewed (commented):
thx, gonna mark this on blocked on #149114

*[View changes since this review](https://triagebot.infra.rust-lang.org/gh-changes-since/rust-lang/rust/150122/ed0006a7ba2dc8fabab8ea94d6f843886311b3c7..7c13df66b421510f0b238249dc5cac3fc24aed0d)*

reddevilmidzy commented:
@rustbot label -A-LLVM

rust-log-analyzer commented:
The job **`aarch64-gnu-llvm-20-2`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/20479895684/job/58851363680) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/58851363680) [(plain)](https://github.com/rust-lang/rust/commit/60c8075efacc161b7e49c72cae00af34d19412dd/checks/58851363680/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
[TIMING:end] tool::ToolBuild { build_compiler: Compiler { stage: 0, host: aarch64-unknown-linux-gnu, forced_compiler: false }, target: aarch64-unknown-linux-gnu, tool: "tidy", path: "src/tools/tidy", mode: ToolBootstrap, source_type: InTree, extra_features: [], allow_features: "", cargo_args: [], artifact_kind: Binary } -- 11.478
[TIMING:end] tool::Tidy { compiler: Compiler { stage: 0, host: aarch64-unknown-linux-gnu, forced_compiler: false }, target: aarch64-unknown-linux-gnu } -- 0.000
fmt check
Diff in /checkout/compiler/rustc_ast_lowering/src/pat.rs:441:
     fn lower_ty_pat_mut(&mut self, pattern: &TyPat, base_type: Span) -> hir::TyPat<'hir> {
         // loop here to avoid recursion
         let pat_hir_id = self.lower_node_id(pattern.id);
-        let node = match &pattern.kind {
-            TyPatKind::Range(e1, e2, Spanned { node: end, span }) => hir::TyPatKind::Range(
-                e1.as_deref().map(|e| self.lower_anon_const_to_const_arg_alloc(e)).unwrap_or_else(|| {
-                    self.lower_ty_pat_range_end(
-                        hir::LangItem::RangeMin,
-                        span.shrink_to_lo(),
-                        base_type,
-                    )
-                }),
-                e2.as_deref()
-                    .map(|e| match end {
---
-                            base_type,
-                        )
-                    }),
-            ),
-            TyPatKind::NotNull => hir::TyPatKind::NotNull,
-            TyPatKind::Or(variants) => {
-                hir::TyPatKind::Or(self.arena.alloc_from_iter(
+        let node =
+            match &pattern.kind {
+                TyPatKind::Range(e1, e2, Spanned { node: end, span }) => hir::TyPatKind::Range(
+                    e1.as_deref()
+                        .map(|e| self.lower_anon_const_to_const_arg_alloc(e))
+                        .unwrap_or_else(|| {
+                            self.lower_ty_pat_range_end(
+                                hir::LangItem::RangeMin,
+                                span.shrink_to_lo(),
+                                base_type,
---
+                                base_type,
+                            )
+                        }),
+                ),
+                TyPatKind::NotNull => hir::TyPatKind::NotNull,
+                TyPatKind::Or(variants) => hir::TyPatKind::Or(self.arena.alloc_from_iter(
                     variants.iter().map(|pat| self.lower_ty_pat_mut(pat, base_type)),
-                ))
-            }
-            TyPatKind::Err(guar) => hir::TyPatKind::Err(*guar),
-        };
+                )),
+                TyPatKind::Err(guar) => hir::TyPatKind::Err(*guar),
+            };
 
         hir::TyPat { hir_id: pat_hir_id, kind: node, span: self.lower_span(pattern.span) }
     }
Diff in /checkout/compiler/rustc_ast_lowering/src/expr.rs:158:
                 }
                 ExprKind::Cast(expr, ty) => {
                     let expr = self.lower_expr(expr);
-                    let ty =
-                        self.lower_ty_alloc(ty, ImplTraitContext::Disallowed(ImplTraitPosition::Cast));
+                    let ty = self
+                        .lower_ty_alloc(ty, ImplTraitContext::Disallowed(ImplTraitPosition::Cast));
                     hir::ExprKind::Cast(expr, ty)
                 }
                 ExprKind::Type(expr, ty) => {
Diff in /checkout/compiler/rustc_ast_lowering/src/expr.rs:166:
                     let expr = self.lower_expr(expr);
-                    let ty =
-                        self.lower_ty_alloc(ty, ImplTraitContext::Disallowed(ImplTraitPosition::Cast));
+                    let ty = self
+                        .lower_ty_alloc(ty, ImplTraitContext::Disallowed(ImplTraitPosition::Cast));
                     hir::ExprKind::Type(expr, ty)
                 }
                 ExprKind::AddrOf(k, m, ohs) => {
Diff in /checkout/compiler/rustc_ast_lowering/src/expr.rs:371:
                     *kind,
                     self.lower_expr(expr),
                     ty.as_ref().map(|ty| {
-                        self.lower_ty_alloc(ty, ImplTraitContext::Disallowed(ImplTraitPosition::Cast))
+                        self.lower_ty_alloc(
+                            ty,
+                            ImplTraitContext::Disallowed(ImplTraitPosition::Cast),
+                        )
                     }),
                 ),
 
fmt: checked 6611 files
```

</details>

reddevilmidzy commented:
https://github.com/rust-lang/rust/pull/149114 merged and rebase completed.

@rustbot ready

lcnr commented:
cc @BoxyUwU 

@bors r+ rollup

bors commented:
:pushpin: Commit 598d01b2d9dbad4e51b1a4bcd42ca061f29cc564 has been approved by `lcnr`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=lcnr 598d01b2d9dbad4e51b1a4bcd42ca061f29cc564 -->
<!-- homu: {"type":"Approved","sha":"598d01b2d9dbad4e51b1a4bcd42ca061f29cc564","approver":"lcnr","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

reddevilmidzy committed:
rename the `lower_ty` and `lower_ty_direct` to `lower_ty_and_alloc` and
`lower_ty`

reddevilmidzy committed:
rename the `lower_anon_const_to_const_arg` and
`lower_anon_const_to_const_arg_direct` to
`lower_anon_const_to_const_arg_and_alloc` and
`lower_anon_const_to_const_arg`
